### PR TITLE
Fix build on Apple Silicon macs

### PR DESCRIPTION
**Pull request by Atemu**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Fixes the build on M1 macs.
For more information see [this comment](https://github.com/TeamNewPipe/NewPipe/pull/7782#issuecomment-1034371070)

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -101,7 +101,7 @@ ext {
     checkstyleVersion = '9.2.1'
 
     androidxLifecycleVersion = '2.3.1'
-    androidxRoomVersion = '2.3.0'
+    androidxRoomVersion = '2.4.1'
```
Comment (mhmdanas) [inline app/build.gradle:104]: Changelog: https://developer.android.com/jetpack/androidx/releases/room

**Review by litetex — CHANGES_REQUESTED** [review]
@Atemu 
Thank you for the PR.

> Fixes the build on M1 macs.

That's not enough.
Please describe how your changes fix the problem.

File: app/build.gradle (hunk 1 above)
Comment (Atemu) [reply app/build.gradle:104]: I'll be running this patch backported to master in the coming days.

**Review by litetex — CHANGES_REQUESTED** [review]
Not sure why my review is requested but the situation since
https://github.com/TeamNewPipe/NewPipe/pull/7782#pullrequestreview-872360482
hasn't changed...

**Comment by mhmdanas** [comment]
@litetex it somewhat did, read the commit messages.

**Comment by litetex** [comment]
> @litetex it somewhat did, read the commit messages.

Thanks for the hint, however 2 problems:
1. A commit message != a comment here
2. "compileSdk 31" is still open and waiting for explanation

**Comment by Atemu** [comment]
@litetex ah, you meant the PR comment! I thought you meant the commit messages.

You did not mention anything regarding compileSdk 31 in your review though, I'm a bit confused.

**Comment by litetex** [comment]
> You did not mention anything regarding compileSdk 31 in your review though, I'm a bit confused.

Well you updated the compileSdk to 31.
Why did you do that?

**Comment by mhmdanas** [comment]
From the commit message:

> Required for newer versions of some dependencies

So I guess it's needed for the dependency update in the other commit.

**Comment by Atemu** [comment]
Should I be more specific about it or even put it in the same commit?

**Comment by litetex** [comment]
Simply answer my questions in a comment next time. The commit messages are great too but they are easily overlooked in a conversation here.

I also still don't know
* what problem with the current build is present on Mac's. Our CI (which uses MacOS) works fine. Is this a problem with Android studio, the emulator, the build or something else?
* how the above changes (updating a dependency) fixed the problem on Mac. I see no fixed bug regarding MacOS in the changelogs.

It would be great too have things like stacktraces, bug reports, links, etc which we can refer to?

**Comment by ktprograms** [comment]
@litetex It's a problem that earlier versions of Android Room can't be used on Arm64 (M1) Macs.

AFAIK, It can be fixed either by updating the Room dependency, or by adding `kapt "org.xerial:sqlite-jdbc:3.34.0"` to the dependencies.

**Comment by ktprograms** [comment]
@litetex Here's where it's mentioned in the Android Room Changelogs: https://developer.android.com/jetpack/androidx/releases/room#2.4.0-alpha03 (first bullet point under the `Bug Fixes` heading).

> Fixed an issue with Room’s SQLite native library to support Apple’s M1 chips. ([b/174695268](https://issuetracker.google.com/issues/174695268)

**Comment by litetex** [comment]
Thx @ktprograms for clarifying that.
Now I understand what is going on here 👍 

----

> Should I really include a link to commit/patchnotes?

👉 
> Simply answer my questions in a comment next time

**Review by litetex — APPROVED** [review]
LGTM

Did a quick test and couldn't find any problems

**Comment by Stypox** [comment]
Warning: the bump to `compileSdk` should be tested carefully on Android 12 and non-12 (devices)/emulators. This this is so simple to revert, I'd merge this after `0.22.0` is released and then test on the `dev` branch if anything wrong happens when updating the compile sdk.

**Pull request merged**